dbsnp-importer/src/main/java/uk/ac/ebi/eva/dbsnpimporter/io/readers/SubSnpCoreFieldsRowMapper.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2017 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.ebi.eva.dbsnpimporter.io.readers;
+
+import org.springframework.jdbc.core.RowMapper;
+
+import uk.ac.ebi.eva.dbsnpimporter.models.SubSnpCoreFields;
+
+import java.math.BigDecimal;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+/**
+ * Maps the database fields that correspond to an SS ID along with its contig and (optionally) chromosome coordinates.
+ *
+ * TODO Add reference allele
+ * TODO Add alternate allele
+ */
+public class SubSnpCoreFieldsRowMapper implements RowMapper<SubSnpCoreFields> {
+
+    public static final String SUBSNP_ID_COLUMN = "ss_id";
+
+    public static final String REFSNP_ID_COLUMN = "rs_id";
+
+    public static final String CONTIG_NAME_COLUMN = "contig_name";
+
+    public static final String CONTIG_START_COLUMN = "contig_start";
+
+    public static final String CONTIG_END_COLUMN = "contig_end";
+
+    public static final String CHROMOSOME_COLUMN = "chromosome";
+
+    public static final String CHROMOSOME_START_COLUMN = "chromosome_start";
+
+    public static final String CHROMOSOME_END_COLUMN = "chromosome_end";
+
+    public static final String SNP_ORIENTATION_COLUMN = "snp_orientation";
+
+    public static final String CONTIG_ORIENTATION_COLUMN = "contig_orientation";
+
+
+    @Override
+    public SubSnpCoreFields mapRow(ResultSet resultSet, int rowNum) throws SQLException {
+        Integer rsId = resultSet.getObject(REFSNP_ID_COLUMN, Integer.class);
+        Long rsIdLong = rsId == null ? null : new Long(rsId);
+
+        Long chrStart = resultSet.getObject(CHROMOSOME_START_COLUMN, Long.class);
+        Integer chrStartLong = chrStart == null ? null : chrStart.intValue();
+
+        BigDecimal chrEnd = resultSet.getObject(CHROMOSOME_END_COLUMN, BigDecimal.class);
+        Integer chrEndLong = chrStart == null ? null : chrEnd.intValue();
```

Comment:
Why all these castings?

1. The one from `Long` to `Integer` could make us lose values in the process.
2. We can just change the types in `SubSnpCoreFields` to match those from the database. And since we are not going to use it as our database model, so we can just cast when it is actually necessary during the conversion.